packages/client/modules/userDashboard/components/GcalModal/GcalModal.tsx

```diff
@@ -0,0 +1,173 @@
+import DateTimePicker from './DateTimePicker'
+import Checkbox from '../../../../components/Checkbox'
+import StyledError from '../../../../components/StyledError'
+import PlainButton from '../../../../components/PlainButton/PlainButton'
+import {Close} from '@mui/icons-material'
+import {CreateGcalEventInput} from '../../../../__generated__/CreateGcalEventMutation.graphql'
+import useForm from '../../../../hooks/useForm'
+import Legitity from '../../../../validation/Legitity'
+
+const Wrapper = styled('div')({
+  display: 'flex',
+  justifyContent: 'flex-end',
+  paddingTop: 16
+})
+
+const StyledInput = styled('input')({
+  border: `1px solid ${PALETTE.SLATE_400}`,
+  borderRadius: 4,
+  color: PALETTE.SLATE_800,
+  fontSize: 16,
+  font: 'inherit',
+  margin: '8px 0',
+  padding: '12px 16px',
+  outline: 0,
+  width: '100%',
+  '::placeholder': {
+    color: PALETTE.SLATE_600
+  }
+})
+
+const StyledDialogContainer = styled(DialogContainer)({
+  width: 'auto'
+})
+
+const CloseIcon = styled(Close)({
+  color: PALETTE.SLATE_600,
+  cursor: 'pointer',
+  '&:hover': {
+    opacity: 0.5
+  }
+})
+
+const StyledCloseButton = styled(PlainButton)({
+  height: 24,
+  marginLeft: 'auto'
+})
+
+const ErrorMessage = styled(StyledError)({
+  textAlign: 'left',
+  paddingBottom: 8
+})
+
+const validateTitle = (title: string) => {
+  return new Legitity(title).trim().min(2, `C’mon, you call that a title?`)
+}
+
+interface Props {
+  handleCreateGcalEvent: (
+    CreateGcalEventInput: Omit<CreateGcalEventInput, 'teamId' | 'meetingId'>
+  ) => void
+  closeModal: () => void
+}
+
+const GcalModal = (props: Props) => {
+  const {handleCreateGcalEvent, closeModal} = props
+
+  const startOfNextHour = dayjs().add(1, 'hour').startOf('hour')
+  const endOfNextHour = dayjs().add(2, 'hour').startOf('hour')
+  const [start, setStart] = useState(startOfNextHour)
+  const [end, setEnd] = useState(endOfNextHour)
+  const [inviteTeam, setInviteTeam] = useState(true)
+
+  const {fields, onChange} = useForm({
+    title: {
+      getDefault: () => ''
+    },
+    description: {
+      getDefault: () => ''
+    }
+  })
+  const titleErr = fields.title.error
+
+  const handleClick = () => {
+    const title = fields.title.value
+    const titleRes = validateTitle(title)
+    if (titleRes.error) {
+      fields.title.setError(titleRes.error)
+      return
+    }
+    const startTimestamp = start.unix()
+    const endTimestamp = end.unix()
+    const description = fields.description.value
+    const timeZone = Intl.DateTimeFormat().resolvedOptions().timeZone
+    const input = {
+      title,
+      description,
+      startTimestamp,
+      endTimestamp,
+      inviteTeam,
+      timeZone
+    }
+    handleCreateGcalEvent(input)
+  }
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    if (titleErr) {
+      fields.title.setError('')
+    }
+    onChange(e)
+  }
+
+  return (
+    <StyledDialogContainer>
```

Comment:
+1 We recently introduced modal dialog based on radix-ui, see `ui/Dialog` folder in `client`. I would refactor or at least create tech debt issue to avoid increasing the legacy code base

Reply:
I looked at using it, but it initially had a transparent background. I added a div within `Dialog` and added a white background, but it didn't have the border-radius or shadow of this modal.

I'd be happy to create a GitHub issue, although it looks like we're only using the Dialog in one place at the moment, so I imagine it would fall into an issue that encapsulates the entire refactor?

Reply:
> so I imagine it would fall into an issue that encapsulates the entire refactor?

This could be a solution but if we keep introducing legacy dialogs, we would need to do more job later
Also, I'm not sure if we need to do this job at once. Probably easier to update a specific place if you started to work on it.

> I looked at using it, but it initially had a transparent background. I added a div within Dialog and added a white background, but it didn't have the border-radius or shadow of this modal.

I'll create an issue and assign it myself to it, because I'm curious to check what is going on and try to use Dialog I created somewhere else

Reply:
Created a PR #8591